### PR TITLE
[MM-40432] Add telemetry for follower notifications

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR sends following events:
1. On retrospective notification sent to the follower, user clicked on retrospective URL - action is `follower_clicked_retrospective_dm`
2. On Run start notification sent to the follower, user clicked on run link(we have a backstage link) - action is `follower_clicked_run_started_dm`
3. On Run finish notification sent to the follower, user clicked on run link - action is `follower_clicked_run_finished_dm`

#### Ticket Link

  Fixes: https://mattermost.atlassian.net/browse/MM-40432

#### Checklist

- [X] Telemetry updated
- ~~[ ] Gated by experimental feature flag~~
- ~~[ ] Unit tests updated~~

### CODE REVIEW

File: server/app/playbook_run_service.go [hunk 1]
```diff
@@ -119,34 +119,46 @@ func (s *PlaybookRunServiceImpl) GetPlaybookRuns(requesterInfo RequesterInfo, op
 	}, nil
 }
 
-func (s *PlaybookRunServiceImpl) buildPlaybookRunCreationMessage(playbookTitle, playbookID string, playbookRun *PlaybookRun, owner *model.User) (string, error) {
+func (s *PlaybookRunServiceImpl) buildPlaybookRunCreationMessage(playbookTitle, playbookID string, playbookRun *PlaybookRun, owner *model.User) (string, string, error) {
```
Comment (crspeller) [inline server/app/playbook_run_service.go:122]: Adding telemetry here really makes a mess of this.

File: webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx [hunk 2]
```diff
@@ -26,6 +28,18 @@ interface Props {
 export const Retrospective = (props: Props) => {
     const allowRetrospectiveAccess = useAllowRetrospectiveAccess();
     const {formatMessage} = useIntl();
+    const url = new URL(window.location.href);
+    const searchParams = new URLSearchParams(url.searchParams);
+
+    if (searchParams.has('telem') && props.playbookRun) {
```
Comment (crspeller) [inline webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx:34]: This is a cool solution but it involves sticking this sort of thing all over the app. Really making components do more then they should. 

I wonder if we can solve this centrally using history.listen from react router?  https://github.com/remix-run/history/blob/main/docs/getting-started.md#listening

**Review by crspeller — COMMENTED** [review]
Concerned about the complexity here.

File: webapp/src/components/backstage/playbook_runs/playbook_run_backstage/playbook_run_backstage.tsx [hunk 3]
```diff
@@ -246,6 +248,19 @@ const PlaybookRunBackstage = () => {
...
+    if (searchParams.has('telem') && match.params.playbookRunId) {
+        const action = searchParams.get('telem');
+        if (action) {
+            telemetryEventForPlaybookRun(match.params.playbookRunId, action);
+        }
+        searchParams.delete('telem');
+        url.search = searchParams.toString();
+        browserHistory.replace({pathname: url.pathname, search: url.search});
+    }
+
```
Comment (cpoile) [inline webapp/src/components/backstage/playbook_runs/playbook_run_backstage/playbook_run_backstage.tsx:263]: Hopefully we can just remove this now:
```suggestion
```
🎉

**Review by cpoile — CHANGES_REQUESTED** [review]

File: webapp/src/components/backstage/playbook_runs/playbook_run_backstage/playbook_run_backstage.tsx (hunk 3 above)
Comment (iomodo) [reply webapp/src/components/backstage/playbook_runs/playbook_run_backstage/playbook_run_backstage.tsx]: removed, tested, worked like a charm. Thanks @cpoile

**Review by cpoile — APPROVED** [review]
🎉

**Review by crspeller — APPROVED** [review]

**Pull request merged**